beanstalk_worker/worker_start_scan.py

```diff
@@ -26,212 +26,340 @@
+        logger.log(
+            level=logging.INFO,
+            msg="Executing atomic scan:  %s" % " ".join(cmd))
+
+        process = subprocess.Popen(
+            cmd,
+            stderr=subprocess.PIPE,
+            stdout=subprocess.PIPE)
+
+        out, err = process.communicate()
+
+        if out != "":
+            # TODO: hacky and ugly. figure a better way
+            # https://github.com/projectatomic/atomic/issues/577
+            output_json_file = os.path.join(
+                out.strip().split()[-1].split('.')[0],
+                "_%s" % image_rootfs_path.split('/')[1],
+                SCANNERS[self.scanner_name]["output_filename"][0]
+            )
 
-        if os.path.exists(output_json_file):
-            json_data = json.loads(open(output_json_file).read())
+            if os.path.exists(output_json_file):
+                json_data = json.loads(open(output_json_file).read())
+            else:
+                logger.log(level=logging.FATAL,
+                           msg="No scan results found at %s" % output_json_file)
+                raise
         else:
-            logger.log(level=logging.FATAL,
-                       msg="No scan results found at %s" % output_json_file)
-            return
-    else:
-        logs = ""
-
-    logger.log(level=logging.INFO,
-               msg="Unmounting image's rootfs from %s" % image_rootfs_path)
-
-    mount_object.unmount()
-
-    os.rmdir(image_rootfs_path)
-
-    logger.log(level=logging.INFO, msg="Removing the image %s" % image_full_name)
-    conn.remove_image(image=image_full_name, force=True)
-
-    logger.log(level=logging.INFO, msg="Finished scan...")
-
-    # if msg != "" and logs != "":
-    if json_data != None:
-        d = {
-            "image": image_full_name,
-            "msg": "Container image requires update",
-            "logs": json.dumps(json_data),
-            "action": "start_delivery",
-            "name_space": namespace,
-            "notify_email": notify_email
+            logs = {}
+
+        self.unmount_image(self.mount_object)
+        self.remove_image(image_rootfs_path, image_under_test)
+        logger.log(level=logging.INFO, msg="Finished test...")
+        return self.process_output(json_data)
+
+    def process_output(self, json_data):
+        """
+        Process the output from the scanner, parse the json and
+        add meaningful information based on validations
+        """
+        data = {
+            "scanner_name": self.scanner_name
         }
-    else:
-        d = {
-            "image": image_full_name,
-            "msg": "No updates required",
-            "logs": "",
-            "action": "start_delivery",
-            "name_space": namespace,
-            "notify_email": notify_email
-        }
-    bs.use("master_tube")
-    jid = bs.put(json.dumps(d))
-    logger.log(
-        level=logging.INFO,
-        msg="Put job on master tube with id: %d" % jid
-    )
+        if not json_data["Package Updates"]:
```

Comment:
I'm a little confused here about when this condition will hold `True`. Wouldn't it evaluate to `True` when `Package Updates` key holds no data? Which is contrary to what we're looking for.

```python
In [17]: d = {}

In [18]: d["updates"] = None

In [19]: d
Out[19]: {'updates': None}

In [21]: if d["updates"]:
    print "if"
else:
    print "else"
   ....: 
else

In [22]: if not d["updates"]:
    print "if"
else:
    print "else"

In [23]: d["updates"] = 123

In [24]: if not d["updates"]:
    print "if"
else:
    print "else"
   ....: 
else

```